Refs #21183 -- feat(admin): add toggle button for list filters

jsuchan-reef opened the pull request:
- reusing some parts of `nav-sidebar` without doing too much unnecessary changes in said element to keep the diff small at this stage of PR.

- forum thread https://forum.djangoproject.com/t/ticket-21183-reopening/25269


before
![before](https://github.com/django/django/assets/135112713/92ec0c5d-bb2a-47ed-af3b-7c0eb21e49de)


after (expanded / collapsed)
![after](https://github.com/django/django/assets/135112713/4db0b30a-2a34-4359-a896-b8b0b4ed72e5)

jsuchan-reef commented on django/contrib/admin/static/admin/css/sidebar.css:
```diff
@@ -51,24 +59,36 @@
 }
 
+.main > #changelist-filter {
+    display: none;
```
using `display` instead of `visibility` (as `#nav-sidebar` does) because it doesnt need additional justifications via in `flex` and `margin` (as `#nav-sidebar` does)

knyghty reviewed (commented):
Thanks for the work here.

I made a first pass, though I didn't look at the code yet at all.

With [`enable_nav_sidebar=False`](https://docs.djangoproject.com/en/4.2/ref/contrib/admin/#django.contrib.admin.AdminSite.enable_nav_sidebar), the open/close button is still visible without styling:

<img width="290" alt="Screenshot 2023-11-24 at 19 03 28" src="https://github.com/django/django/assets/3871354/2015c89f-6f2b-4dc7-8bca-a2665387ed9e">

It also looks like it should be openable and closable because of the full height, which may or may not be okay.

A horizontal scrollbars appear at all screen widths <= 1185px (Mac OS, Firefox). Example at tablet size:

<img width="1122" alt="Screenshot 2023-11-24 at 19 13 01" src="https://github.com/django/django/assets/3871354/782977f7-2137-44c6-9206-1a3f79543874">

This makes the mobile experience unusable:

https://github.com/django/django/assets/3871354/f66cfcf8-e576-4ceb-99ec-4eb88595d215

When you do scroll to the right, the button obscures the sidebar:

<img width="1103" alt="Screenshot 2023-11-24 at 19 13 47" src="https://github.com/django/django/assets/3871354/fa89d938-5331-49cf-826e-80df4cf22789">

This really needs to be fixed for the bigger screen sizes. For smaller screen sizes, it may be best to place it underneath as currently happens.

Two things we need to clear up:

1. Should this be independent of `enable_nav_sidebar`?
2. What should the bahaviour be on mobile?

knyghty commented:
One other point of UX:

For the nav sidebar, if you have the sidebar closed, you still know where you are because there are breadcrumbs and headers and so.

For the filter sidebar, there is no way to know if there are filters applied or not when the sidebar is closed, other than looking at the URL in the address bar. This seems like an oversight. I think there should be some indicator for the user that they are viewing a filtered list.

jsuchan-reef commented:
thanks for the feedback, I'll look into it.

jsuchan-reef commented:
@knyghty I believe I addressed all the issues mentioned, although I couldnt reproduce some of them (maybe fixed as a byproduct of some other changes?).

mjurbanski-reef commented:
@knyghty anything we can do to get this moving along?

knyghty commented:
> @knyghty anything we can do to get this moving along?

Before anyone is going to review this it probably needs:
- To not be in draft
- To have an accepted ticket (I don't think referencing a wontfix ticket is enough here)
- New tests, not just fixing the existing ones
- It will also need selenium tests
- And at least a release note, not sure personally about other docs.

It's also not clear to me if you addressed the point I made above about indicating if the page is filtered.
